diff -ur cpu2006/benchspec/CPU2006/458.sjeng/src/leval.c cpu2006.mod/benchspec/CPU2006/458.sjeng/src/leval.c
--- cpu2006/benchspec/CPU2006/458.sjeng/src/leval.c	2016-11-05 14:37:20.354181074 +0530
+++ cpu2006.mod/benchspec/CPU2006/458.sjeng/src/leval.c	2016-11-05 14:38:58.266183666 +0530
@@ -25,7 +25,7 @@
 0,0,0,0,0,0,0,0,0,0,0,0,
 0,0,0,0,0,0,0,0,0,0,0,0};
 
-static int l_bishop_mobility(int square)
+/*static */int l_bishop_mobility(int square)
 {
   register int l;
   register int m = 0;
@@ -42,7 +42,7 @@
   return m;
 }
 
-static int l_rook_mobility(int square)
+/*static */int l_rook_mobility(int square)
 {
   register int l;
   register int m = 0;
@@ -60,7 +60,7 @@
 }
 
 
-static int l_knight_mobility(int square)
+/*static */int l_knight_mobility(int square)
 { 
   static const int knight_o[8] = {10, -10, 14, -14, 23, -23, 25, -25};
   register int d, m = 0;
@@ -73,7 +73,7 @@
   return m;
 }
 
-static int l_pawn_mobility(int square)
+/*static */int l_pawn_mobility(int square)
 {
   register int m = 0;
 
@@ -89,7 +89,7 @@
   return m;
 }
 
-static int l_king_mobility(int square)
+/*static */int l_king_mobility(int square)
 {
   static const int king_o[8] = {13, 12, 11, 1, -1, -11, -12, -13};
   register int d, m = 0;
diff -ur cpu2006/benchspec/CPU2006/458.sjeng/src/neval.c cpu2006.mod/benchspec/CPU2006/458.sjeng/src/neval.c
--- cpu2006/benchspec/CPU2006/458.sjeng/src/neval.c	2016-11-05 14:37:20.346181074 +0530
+++ cpu2006.mod/benchspec/CPU2006/458.sjeng/src/neval.c	2016-11-05 14:39:29.906184503 +0530
@@ -202,7 +202,7 @@
 
 }
 
-static int bishop_mobility(int square)
+/*static */int bishop_mobility(int square)
 {
   register int l;
   register int m = 0;
@@ -217,7 +217,7 @@
   return m;
 }
 
-static int rook_mobility(int square)
+/*static */int rook_mobility(int square)
 {
   register int l;
   register int m = 0;
diff -ur cpu2006/benchspec/CPU2006/458.sjeng/src/seval.c cpu2006.mod/benchspec/CPU2006/458.sjeng/src/seval.c
--- cpu2006/benchspec/CPU2006/458.sjeng/src/seval.c	2016-11-05 14:37:20.354181074 +0530
+++ cpu2006.mod/benchspec/CPU2006/458.sjeng/src/seval.c	2016-11-05 14:40:16.250185729 +0530
@@ -30,7 +30,7 @@
 static const int knight_o[8] = {10, -10, 14, -14, 23, -23, 25, -25};
   static const int king_o[8] = {13, 12, 11, 1, -1, -11, -12, -13};
 
-static int s_bishop_mobility(int square)
+/*static */int s_bishop_mobility(int square)
 {
   register int l;
   register int m = 0;
@@ -47,7 +47,7 @@
   return m << 2;
 }
 
-static int s_rook_mobility(int square)
+/*static */int s_rook_mobility(int square)
 {
   register int l;
   register int m = 0;
@@ -64,7 +64,7 @@
   return m << 2;
 }
 
-static int s_knight_mobility(int square)
+/*static */int s_knight_mobility(int square)
 { 
   register int d, m = 0;
   
@@ -76,7 +76,7 @@
   return m << 2;
 }
 
-static int s_pawn_mobility(int square)
+/*static */int s_pawn_mobility(int square)
 {
   register int m = 0;
 
@@ -92,7 +92,7 @@
   return m << 3;
 }
 
-static int s_king_mobility(int square)
+/*static */int s_king_mobility(int square)
 {
   register int d, m = 0;
   
@@ -104,7 +104,7 @@
   return m << 2;
 }
 
-static int black_saccers(int square)
+/*static */int black_saccers(int square)
 {
   register int f = FALSE;
 
@@ -183,7 +183,7 @@
   
 }
 
-static int white_saccers(int square)
+/*static */int white_saccers(int square)
 {
   /* for black pawn on square, any black
    * pieces that can sac themselves to it? */
