diff -ur llvm-3.6.0/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp llvm-3.6.0.mod/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp
--- llvm-3.6.0/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp	2016-06-12 16:04:18.189375785 +0530
+++ llvm-3.6.0.mod/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp	2016-06-12 16:59:20.409511630 +0530
@@ -2243,8 +2243,9 @@
 }
 
 static inline bool areJTsAllowed(const TargetLowering &TLI) {
-  return TLI.isOperationLegalOrCustom(ISD::BR_JT, MVT::Other) ||
-         TLI.isOperationLegalOrCustom(ISD::BRIND, MVT::Other);
+  return false;
+//  return TLI.isOperationLegalOrCustom(ISD::BR_JT, MVT::Other) ||
+//         TLI.isOperationLegalOrCustom(ISD::BRIND, MVT::Other);
 }
 
 static APInt ComputeRange(const APInt &First, const APInt &Last) {
